lib/assert.ts

```diff
@@ -28,18 +28,24 @@ import path from 'node:path';
 import {isString} from '../shared/common-utils.js';
 import {parse} from '../shared/stacktrace.js';
 
+let ce_base_directory = '';
+
 const filePrefix = 'file://';
 
+export function setBaseDirectory(base_url_path: URL) {
+    ce_base_directory = base_url_path.pathname;
+}
+
 function removeFileProtocol(path: string) {
     if (path.startsWith(filePrefix)) {
         return path.slice(filePrefix.length);
     }
     return path;
 }
 
-function check_path(parent: URL, directory: string) {
+function check_path(parent: string, directory: string) {
     // https://stackoverflow.com/a/45242825/15675011
-    const relative = path.relative(parent.pathname, directory);
+    const relative = path.relative(parent, directory);
```

Comment:
we used to pass "parent.pathname" here (it was a URL); and now I just store the string (as we don't care about it being a URL as far as I can tell)

Reply:
Oh I see, parent is always the base dir. All good it seems :)